pkg/ui/app/components/graphs.tsx

```diff
@@ -77,59 +99,270 @@ export class Axis extends React.Component<AxisProps, {}> {
+}
+
+function ComputeCountAxisDomain(
+  min: number, max: number, tickCount: number
+): AxisDomain {
+  let range = max - min;
+  let increment = computeNormalizedIncrement(range, tickCount);
+  let axisDomain = new AxisDomain(min, max, increment);
+  if (Math.floor(increment) !== increment) {
+      axisDomain.tickFormat = d3.format(".1f");
+  } else {
+      axisDomain.tickFormat = d3.format("s");
+  }
+  axisDomain.guideFormat = d3.format(".4s");
+  axisDomain.label = "count";
+  return axisDomain;
+}
+
+const byteLabels = ["bytes", "kilobytes", "megabytes", "gigabytes", "terabytes", "petabytes", "exabytes", "zettabytes", "yottabytes"];
+
+function ComputeByteAxisDomain(
+  min: number, max: number, tickCount: number
+): AxisDomain {
+  // Compute an appropriate unit for the maximum value to be displayed.
+  let prefixExponent = ComputePrefixExponent(max, 1024, byteLabels);
+  let prefixFactor = Math.pow(1024, prefixExponent);
+
+  // Compute increment on min/max after conversion to the appropriate prefix unit.
+  let increment = computeNormalizedIncrement(max / prefixFactor - min / prefixFactor, tickCount);
+
+  // Create axis domain by multiplying computed increment by prefix factor.
+  let axisDomain = new AxisDomain(min, max, increment * prefixFactor);
+
+  // Apply the correct label to the axis.
+  axisDomain.label = byteLabels[prefixExponent];
+
+  // Format ticks to display as the correct prefix unit.
+  let unitFormat: (v: number) => string;
+  if (Math.floor(increment) !== increment) {
+      unitFormat = d3.format(".1f");
+  } else {
+      unitFormat = d3.format("s");
+  }
+  axisDomain.tickFormat = (v: number) => {
+    return unitFormat(v / prefixFactor);
+  };
+
+  axisDomain.guideFormat = Bytes;
+  return axisDomain;
+}
+
+const durationLabels = ["nanoseconds", "microseconds", "milliseconds", "seconds"];
+
+function ComputeDurationAxisDomain(
+  min: number, max: number, tickCount: number
+): AxisDomain {
+  let prefixExponent = ComputePrefixExponent(max, 1000, durationLabels);
+  let prefixFactor = Math.pow(1000, prefixExponent);
+
+  // Compute increment on min/max after conversion to the appropriate prefix unit.
+  let increment = computeNormalizedIncrement(max / prefixFactor - min / prefixFactor, tickCount);
+
+  // Create axis domain by multiplying computed increment by prefix factor.
+  let axisDomain = new AxisDomain(min, max, increment * prefixFactor);
+
+  // Apply the correct label to the axis.
+  axisDomain.label = durationLabels[prefixExponent];
+
+  // Format ticks to display as the correct prefix unit.
+  let unitFormat: (v: number) => string;
+  if (Math.floor(increment) !== increment) {
+      unitFormat = d3.format(".1f");
+  } else {
+      unitFormat = d3.format("s");
+  }
+  axisDomain.tickFormat = (v: number) => {
+    return unitFormat(v / prefixFactor);
+  };
+
+  axisDomain.guideFormat = Duration;
+  return axisDomain;
+}
+
+const percentIncrementTable = [0.25, 0.5, 0.75, 1.0];
```

Comment:
Hmm I feel like we don't need to be so restrictive for percents, but we can see how this looks.